### PR TITLE
Respect existing handlers when passed in a TChannel

### PR DESCRIPTION
**Pull request by prashantv**
The user may want to register non-YARPC methods (and the channel may
have some meta/introspection methods) that we should respect.

cc @yarpc/golang

### CODE REVIEW

File: transport/tchannel/inbound.go [hunk 1]
```diff
@@ -56,7 +56,9 @@ type inbound struct {
 }
 
 func (i *inbound) Start(h transport.Handler) error {
-	i.ch.GetSubChannel(i.ch.ServiceName()).SetHandler(handler{h})
+	sc := i.ch.GetSubChannel(i.ch.ServiceName())
+	existing := sc.GetHandlers()
+	sc.SetHandler(handler{existing, h})
```
Comment (abhinav) [inline transport/tchannel/inbound.go]: Mention this fancy behavior somewhere in the documentation for this transport?

**Comment by abhinav** [comment]
Cool! :shipit:

**Comment by breerly** [comment]
This is desirable, ty sir. 🐑

File: transport/tchannel/handler.go [hunk 2]
```diff
@@ -71,10 +71,16 @@ func (c tchannelCall) Response() inboundCallResponse {
...
 // handler wraps a transport.Handler into a TChannel Handler.
 type handler struct {
-	Handler transport.Handler
+	existing map[string]tchannel.Handler
+	Handler  transport.Handler
 }
 
 func (h handler) Handle(ctx context.Context, call *tchannel.InboundCall) {
+	if m, ok := h.existing[call.MethodString()]; ok {
+		m.Handle(ctx, call)
```
Comment (breerly) [inline transport/tchannel/handler.go]: this block is super clean, thx P

File: transport/tchannel/inbound_test.go [hunk 3]
```diff
@@ -96,3 +98,31 @@ func TestInboundInvalidAddress(t *testing.T) {
 	i := NewInbound(ch, ListenAddr("not valid"))
 	assert.Error(t, i.Start(new(transporttest.MockHandler)))
 }
+
+func TestInboundExistingMethods(t *testing.T) {
+	// Create a channel with an existing "echo" method.
+	ch, err := tchannel.NewChannel("foo", nil)
+	require.NoError(t, err)
+	json.Register(ch, json.Handlers{
+		"echo": func(ctx json.Context, in map[string]string) (map[string]string, error) {
```
Comment (breerly) [inline transport/tchannel/inbound_test.go]: suggest renaming `in` to `req` here

Comment (prashantv) [inline transport/tchannel/inbound_test.go]: 👍

File: transport/tchannel/inbound.go (hunk 1 above)
Comment (prashantv) [inline transport/tchannel/inbound.go]: Good point, added a comment to the `NewInbound` method doc

**Pull request merged**